Vehicle data refactoring

atiwari9 opened the pull request:
This proposal is to refactor current vehicle data items to accommodate new vehicle data items and to remove redundancy due to new vehicle data items added in [Vehicle Data Additions proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0253-Vehicle-data-additions.md).

atiwari9 commented:
@theresalech - Ready for review

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
```diff
@@ -0,0 +1,598 @@
+
+## Alternatives considered
+
+* Instead of adding `hazardLights`  to `TurnSignal`, it can be a new vehicle data items in itself.
```
```suggestion
* Instead of adding `hazardLights` to `TurnSignal`, it can be a new vehicle data item in itself.
```

atiwari9 replied:
Done, i am going to move TurnSignal to new proposal.

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
````diff
@@ -0,0 +1,598 @@
+
+```
+
+##### Update function `SubscribeVehicleData` request: 
````
Can you please remove references to the HMI_API that are not changed by this proposal? It will help to make this proposal more concise and readable for SDLC members.

atiwari9 replied:
I have only included the changes already, is there any specific reference you are pointing to.

theresalech replied:
With `TurnSignal` being removed, this comment is no longer valid.  Thanks!

atiwari9 commented on proposals/NNNN-Vehicle-data-refactoring.md:
```diff
@@ -0,0 +1,598 @@
+* Proposal: [SDL_NNNN](NNNN-Vehicle-data-refactoring.md)
+* Author: [Ankur Tiwari](https://github.com/atiwari9)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core | HMI | SDL SERVER | PROXY]
```
Done

atiwari9 commented on proposals/NNNN-Vehicle-data-refactoring.md:
```diff
@@ -0,0 +1,598 @@
+
+## Introduction
+
+This proposal is to refactor current vehicle data items to accommodate new vehicle data items and to remove redundancy due to new vehicle data items added in [Vehicle Data Additions proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0253-Vehicle-data-additions.md).
```
Done.

atiwari9 commented on proposals/NNNN-Vehicle-data-refactoring.md:
````diff
@@ -0,0 +1,598 @@
+</struct>
+```
+
+##### Add to function `SubscribeVehicleData` request: 
````
I have only included the changes already, is there any specific reference you are pointing to.

theresalech replied:
With `TurnSignal` being removed, this comment is no longer valid.  Thanks!

atiwari9 commented on proposals/NNNN-Vehicle-data-refactoring.md:
```diff
@@ -0,0 +1,598 @@
+
+## Motivation
+
+In order to partner with more diverse app partners, we need to provide additional set of vehicle data item through SDL. This goes in line with commitment to enhance SDL with even more rich vehicle data content.
```
Done

atiwari9 commented:
@theresalech  - I have updated the proposal as per your comments. Also i have removed `TurnSignal` changes and moved those to new proposal: [Update vehicle data TurnSignal](https://github.com/smartdevicelink/sdl_evolution/pull/840)

All the API references are for true changes, so there are no redundant references.

Since this proposal is dependent on  [New vehicle data ClimateData](https://github.com/smartdevicelink/sdl_evolution/pull/838) and [New vehicle data GearStatus](https://github.com/smartdevicelink/sdl_evolution/pull/837)  proposals, you may review it for errors etc and then we can park the PR here until [New vehicle data ClimateData](https://github.com/smartdevicelink/sdl_evolution/pull/838) and [New vehicle data GearStatus](https://github.com/smartdevicelink/sdl_evolution/pull/837)  are voted upon.

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
````diff
@@ -0,0 +1,424 @@
+	</history>
+</param>
+```
+
````
Can this section be re-organized to be a bit cleaner and easier to read?  Perhaps group all requests together and all responses together?

For example:

Add the following parameter to these function requests:
- `SubscribeVehicleData`
- `UnsubscribeVehicleData`
...

```xml
<param name="externalTemperature" type="Boolean" mandatory="false" since="2.0" until="X.x">
	<description>The external temperature in degrees celsius</description>
	<history>
		<param name="externalTemperature" type="Boolean" mandatory="false"/>
	</history>
</param>
```
Add the following parameter to these function responses:
- `SubscribeVehicleData`
- `UnsubscribeVehicleData`
...

```xml
<param name="externalTemperature" type="VehicleDataResult" mandatory="false" since="2.0" until="X.x">
	<description>The external temperature in degrees celsius</description>
	<history>
		<param name="externalTemperature" type="VehicleDataResult" mandatory="false"/>
	</history>
</param>
```

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
````diff
@@ -0,0 +1,424 @@
+-<param name="externalTemperature" type="Float" minvalue="-40" maxvalue="100" mandatory="false">
+-	<description>The external temperature in degrees celsius</description>	
+-</param>
+```
````
Same suggestion as above, regarding re-organizing for readability.

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
````diff
@@ -0,0 +1,424 @@
+	</history>
+</param>
+```
+
````
Same suggestion as above, regarding re-organizing for readability.

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
````diff
@@ -0,0 +1,424 @@
+-<param name="prndl" type="Common.PRNDL" mandatory="false">
+-	<description>See PRNDL</description>	
+-</param>
+```
````
Same suggestion as above, regarding re-organizing for readability.

theresalech commented on proposals/NNNN-Vehicle-data-refactoring.md:
```diff
@@ -0,0 +1,424 @@
+	<element name="NEUTRAL">
+		<description>No gear</description>
+	</element>
+	<element name="DRIVE">
```
Please add description

atiwari9 commented:
@theresalech - This PR is no longer needed, so i am going to close it.
